test/lib/yarn.test.ts

```diff
@@ -81,6 +81,27 @@ const SCENARIOS_WITH_FILES = [
     workspace: 'missing-name',
     includeDev: false,
   },
+  {
+    name:
+      'Parse yarn.lock with deeply out of sync app because of local workspaces (root)',
+    workspace: 'out-of-sync-app',
+    includeDev: false,
+    strict: false,
+  },
+  {
+    name:
+      'Parse yarn.lock with deeply out of sync app because of local workspaces (app)',
```

Comment:
do you think that we can make name a little more descriptive?
```suggestion
      'Parse gracefully yarn.lock with local workspaces (app) packages that are not referenced in lock file if scriptOutOfSync is set to false',
```